actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -82,11 +81,19 @@ def format(formatter, filter = nil)
 
       private
 
-      def filter_routes(filter)
-        if filter
+      def filter_routes(options)
```

Comment:
Yeah, exactly, assigning the pattern to each route default and then matching on that.

Sure, it was made with the assumption that people won't really use `-g` and `-c` together. I can't really see a case where that makes sense, do you have any? :smile:

I think we could influence the behavior by how we frame the options in the documentation. I.e. saying `-c` is for when you have a specific controller in mind, and `-g` is for a general search. That way people would be less likely to mash them together.